### PR TITLE
Expanding automated exploratory differential expression calculations (SCP-5621)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND & CHANGES
This update expands the annotation eligibility for automated differential expression calculations to cover "raw" annotations.  These are annotations that are potentially the output from popular clustering algorithms, such as Seurat, Leiden, Louvain, etc.   The belief is that these annotations (like cell type analog annotations) carry a lower risk of false positives when calculating differential expression, and the benefit of extending exploratory differential expression results to a much larger cohort of annotations outweighs any potential risks.  This will also cover any annotations that use the phrase `clust` in the name, following the same logic as above.

An analysis of available data on production showed that including these new annotation will potentially add 174 new studies and 1,408 result sets.  Using our existing differential expression ingests as a benchmark (820 jobs in 2023 at a cost of $77.57), the projected cost is ~$133 to backfill these results.  Our current success rate for automated differential expression ingest is ~72%, which would mean that we can expect approx. 1,013 new result sets.  Given that we have 432 active results across 154 studies now, this would constitute an increase of well over 2x in total results while roughly doubling the number of studies.

Due to technological constraints on processing background jobs, we can't add these results via an automated migration.  However, there is the new `DifferentialExpressionService.backfill_new_results` method that will detect and validate any newly eligible annotations across an entire instance of SCP and launch ingest processes.  This can be done manually from the Rails console at any time.  The method logs progress while validating jobs, and upon completion will output statistics on how many total jobs were run, as well as new results per study (this does not account for failures, only submissions).

Example truncated output:
```
SCP13 has annotations eligible for DE; validating inputs
Checking DE job for SCP13: Major Cell Types (CLUSTER--group--study)                                                                         
  Skipping DE job for SCP13: Major Cell Types (CLUSTER--group--study) due to: SCP13 has no parsed raw counts data                           
Checking DE job for SCP13: Major Cell Types (SUB-CLUSTER--group--study)                                                                     
  Skipping DE job for SCP13: Major Cell Types (SUB-CLUSTER--group--study) due to: SCP13 has no parsed raw counts data                       
...

SCP66 has annotations eligible for DE; validating inputs
Checking DE job for SCP66: Epithelial Cells UMAP (Epithelial Cell Subclusters--group--cluster)
==> DE job for SCP66: Epithelial Cells UMAP (Epithelial Cell Subclusters--group--cluster) successfully launched
SCP66 yielded 1 differential expression jobs; 0 skipped
SCP72 has annotations eligible for DE; validating inputs
Checking DE job for SCP72: Cluster 1 (seurat_clusters--group--study)
==> DE job for SCP72: Cluster 1 (seurat_clusters--group--study) successfully launched
SCP72 yielded 1 differential expression jobs; 0 skipped
...

Total new backfill jobs: 5 across 4 studies
=> {"SCP66"=>1, "SCP72"=>1, "SCP82"=>2, "SCP105"=>1, :total_jobs=>5}
```

Also, this update adds a new workspace (`main-validation-study-hnds4`) to the list of "corrupted" test Terra workspaces.  This will prevent automated cleanup of the affected workspace to prevent CI jobs from hanging on cleanup after running all tests.

#### MANUAL TESTING
1. Boot all services, including `Delayed::Job` workers
2. Open a Rails console session and initiate the backfill process:
```
DifferentialExpressionService.backfill_new_results
```
3. Note log output like above (you will see many studies fail validation for various reasons, like not having raw counts data, or only 1 label for an annotation)
4. You should see a job launch for the `Human milk - differential expression` study on the `Epithelial Cells UMAP (Epithelial Cell Subclusters--group--cluster)` cluster/annotation
5. Wait for the DE jobs to finish computing for the above study (this should take 6-8 minutes)
6. Open your copy of this study in your local instance, and select the `Epithelial Cells UMAP` cluster and `Epithelial Cell Subclusters` annotation
7. Click `Differential expression`, and confirm you see groups for this annotation
8. Select `Cycling Lactocytes` and load the genes table
9. Click `STMN1` and confirm you get a plot like below:
Screenshot 2024-05-02 at 10 39 52 AM

### CODE REVIEW

File: app/lib/differential_expression_service.rb [hunk 1]
```diff
@@ -1,9 +1,14 @@
 # handle launching differential expression ingest jobs
 class DifferentialExpressionService
   extend Loggable
-
-  # regex for matching annotation names for possible cell type analogs
+  # possible cell type analogs
   CELL_TYPE_MATCHER = /cell.*type/i
+  # possible clustering algorithm results
+  CLUSTERING_MATCHER = /(clust|seurat|leiden|louvain|_snn_res)/i
```
Comment (jlchang) [inline app/lib/differential_expression_service.rb:7]: In browsing the breadth of clustering names in the prod database, there was a set of clusterings named SNN_RES_0.XX (where XX was a series of different numbers). I realized that the leading underscore for snn_res would unnecessarily exclude that set of clusterings. Do you have any reservations about removing the underscore for that part of the CLUSTERING_MATCHER?

Comment (bistline) [reply app/lib/differential_expression_service.rb:7]: Nope - I defer to you on that.

Comment (bistline) [reply app/lib/differential_expression_service.rb]: [2955df0](https://github.com/broadinstitute/single_cell_portal_core/pull/2029/commits/2955df088726df4d617406a3378b810d53ec6377)

**Review by jlchang — APPROVED** [review]
Many thanks for the regex edit. Functional review - local backfill worked as described.

Looking forward to increasing DE availability in the portal for these probable cell type-related annotations.

**Review by eweitz — APPROVED** [review]
Code looks good.  Enabling DE exploration in 2x the studies is exciting!  

This seems among SCP's highest value-to-effort work.  Excellent analysis of cost and feature coverage.  

[Context in Slack](https://broadinstitute.slack.com/archives/CBEHTH601/p1714412054573819) has more background details, impact measures, and future ideas.

**Pull request merged**